plugins/woocommerce/src/Blocks/BlockTypes/BlockifiedProductDetails.php

```diff
@@ -32,6 +35,67 @@ protected function get_block_type_style() {
 	 * @return string Rendered block output.
 	 */
 	protected function render( $attributes, $content, $block ) {
-		return $content;
+		return $this->hide_empty_accordion_items( $content, $block );
+	}
+
+	/**
+	 * Hide empty accordion items.
+	 *
+	 * @param string $content Block content.
+	 * @param WP_Block $block Block instance.
+	 *
+	 * @return string Rendered block output.
+	 */
+	private function hide_empty_accordion_items( $content, $block ) {
+		$accordion_items = $this->find_accordion_items( $block->parsed_block );
+
+		if ( ! $accordion_items ) {
+			return $content;
+		}
+
+		$accordion_items_visibility = array_map(
+			function ( $item ) use ( $block ) {
+				$content_block          = end( $item['innerBlocks'] );
+				$rendered_content_block = ( new WP_Block( $content_block, $block->context ) )->render();
+				$p                      = new WP_HTML_Tag_Processor( $rendered_content_block );
+
+				return $p->next_tag( 'img' ) || $p->next_tag( 'video' ) || $p->next_tag( 'iframe' ) || ! empty( wp_strip_all_tags( $rendered_content_block, true ) );
+			},
+			$accordion_items
+		);
+
+		$p = new WP_HTML_Tag_Processor( $content );
+
+		$counter = 0;
+		while ( $p->next_tag( array( 'class_name' => 'wp-block-woocommerce-accordion-item' ) ) ) {
+			if ( ! $accordion_items_visibility[ $counter ] ) {
+				$p->set_attribute( 'style', 'display:none;' );
```

Comment:
Should we also set the [`hidden` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/hidden) for a11y reasons?

Reply:
Good point! Sure!